### DISCUSSION TITLE
a question about subdomains on github pages

### DISCUSSION THREAD

**Discussion by ghost** [community/community#64133] [New to GitHub] [answered]
hello, im new here, i hope i put this in the right place. i have a question regarding github pages:

i would like to page a sort of "subsite" on my github.io website. i know the `https://repository.github.io/subsite/index.html` method, but what id like to do is this: `https://subsite.repository.github.io/` . is this possible?

i did my best to search for existing questions like this, but i didn't find any at all. admittedly, i don't actually know what it's called when a website does this.

edit: subdomain. it's a subdomain, and i was simply being a bit dumb earlier.

**Comment by kyleecodes** [comment, marked as answer]
Hi and welcome! 👋 

By "subsite" do you mean [subdomain](https://en.wikipedia.org/wiki/Subdomain)? Here's the definition from Wikipedia:

> In the Domain Name System (DNS) hierarchy, a subdomain is a domain that is a part of another (main) domain. For example, if a domain offered an online store as part of their website `example.com`, it might use the subdomain `shop.example.com`. 

I believe this is not possible without a custom domain because you need access to the DNS github.io settings, more info here: https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites 

But with a custom domain, it is possible: https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages#using-a-subdomain-for-your-github-pages-site

Hope this helps! 😀

**Reply by ghost** [reply]
this was very helpful, kylee, thank you! i ended up buying a very cheap custom domain, and i successfully pointed my github pages website to it. (and i did definitely mean a subdomain!)

however, im having some trouble understanding the guide to add a subdomain above. these are all intended for people who aren't beginners – which makes sense, but means they tend to use language i find confusing. im finding this to be an issue with the guides from my domain provider as well, namecheap.

i currently own `sitename.online`. to have a custom subdomain like `blog.sitename.online`, where would i set up the subdomain on github's side? would it just go in my existing CNAME file, underneath `sitename.online`?

also, if you know and could explain to me how to set up the subdomain with namecheap, it would be appreciated, but i definitely understand if you can't!
